src/ert/_c_wrappers/enkf/model_config.py

```diff
@@ -1,299 +1,122 @@
-                obs_config = os.path.realpath(obs_config)
-
-            # TIME_MAP_KEY
-            time_map = None
-            time_map_file = config_dict.get(ConfigKeys.TIME_MAP)
-            if time_map_file is not None and not os.path.isfile(
-                os.path.realpath(time_map_file)
-            ):
-                raise ValueError("Error: Time map is not a file")
-            if time_map_file is not None:
-                time_map = TimeMap()
-                time_map.fload(filename=os.path.realpath(time_map_file))
-
-            # GEN_KW_EXPORT_NAME_KEY
-            gen_kw_export_name = config_dict.get(ConfigKeys.GEN_KW_EXPORT_NAME)
-
-            # HISTORY_SOURCE_KEY
-            history_source = config_dict.get(
-                ConfigKeys.HISTORY_SOURCE, HistorySourceEnum.REFCASE_HISTORY
-            )
-
-            c_ptr = self._alloc_full(
-                max_resample,
-                num_realizations,
-                run_path,
-                data_root,
-                ens_path,
-                job_name,
-                obs_config,
-                time_map,
-                gen_kw_export_name,
-                history_source,
-                refcase,
-            )
-
-            if time_map is not None:
-                time_map.convertToCReference(None)
-
-        if c_ptr is None:
-            raise ValueError("Failed to construct ModelConfig instance.")
-
-        super().__init__(c_ptr, is_reference=is_reference)
-
-    def get_history_source(self) -> HistorySourceEnum:
-        return self._get_history_source()
-
-    def set_history_source(
-        self, history_source: HistorySourceEnum, refcase: EclSum
-    ) -> bool:
-        assert isinstance(history_source, HistorySourceEnum)
-        assert isinstance(refcase, EclSum)
-        return self._select_history(history_source, refcase)
-
-    def get_max_internal_submit(self) -> int:
-        return self._get_max_internal_submit()
-
-    def set_max_internal_submit(self, max_value):
-        self._get_max_internal_submit(max_value)
-
-    def getRunpathAsString(self) -> str:
-        return self._get_runpath_as_char()
-
-    def selectRunpath(self, path_key) -> bool:
-        return self._select_runpath(path_key)
-
-    def setRunpath(self, path_format):
-        self._set_runpath(path_format)
-
-    def free(self):
-        self._free()
-
-    def getGenKWExportName(self) -> str:
-        return self._gen_kw_export_name()
-
-    def runpathRequiresIterations(self) -> bool:
-        return self._runpath_requires_iterations()
+        self.num_realizations = num_realizations
+        self.refcase = refcase
+        if config_dir is None:
+            config_dir = os.getcwd()
+        self._config_dir = config_dir
+        self.data_root = data_root if data_root is not None else config_dir
+        self.ens_path: str = (
+            ens_path
+            if ens_path is not None
+            else os.path.join(config_dir, self.DEFAULT_ENSPATH)
+        )
```

Comment:
If you want a shorter version of this piece of code you can try:
``` python 
self.ens_path: str = ens_path or os.path.join(config_dir, self.DEFAULT_ENSPATH)
```
but this might be a bit harder to understand at first glance if you are not used to it so no pressure to change it.

Reply:
oh i like it, thank you! but it will pick the alternative not only when `ens_path` is `None`, but also when it's falsy in the sense of being the empty string. that might not be intended behaviour